### ISSUE TITLE
bug: The function `parse_sql` fails when `COUNT(*)` is in the query

### ISSUE THREAD

**Issue by jpohanka** [ibis-project/ibis#9267]
### What happened?

I have the following toy example of an unbounded table:

```
tx_schema = {
    "date_id": "date",
    "store_id": "int32",
    "sold_amount": "double",
    "quantity": "double",
}

tx = ibis.table(
    name="tx",
    schema=tx_schema
)
```

When I want parse the query `SELECT COUNT(*) AS row_count FROM tx`, it throws a `Keyword: ''` error, as was already reported here: https://github.com/ibis-project/ibis/issues/8597#issuecomment-1986249860

### What version of ibis are you using?

9.0.0

### What backend(s) are you using, if any?

None.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
@jpohanka Thanks for opening the issue!

Would you mind elaborating a bit on what you're using `parse_sql` for? Would love to hear about use cases in the wild.

**Comment by jpohanka** [comment]
@cpcloud Currently just testing the functionality, but I could imagine some migrations of BI reports using raw SQL to Ibis expressions.

**Comment by cpcloud** [comment]
Roger that! It's definitely got a lot of potential.

**Comment by jpohanka** [comment]
@cpcloud I agree, among many things, it could serve as a common abstraction layer for creating SQL expressions over several database systems.
